### DISCUSSION TITLE
Pagination Bullets not rendering with SSR

### DISCUSSION THREAD

**Discussion by myadaval2** [nolimits4web/swiper#5298] [Q&A]
I'm working on SSR using swiper, and I can't get the pagination bullets to show up on the server html. When I build for web targets, the bullets show up correctly when CSR, but when I build for node, the bullets don't show up anymore. If I manually add them in on the server, they get overwritten (erased) when swiper renders on the client (ReactDOM.hydrate). Is this expected to happen?

**Comment by myadaval2** [comment]
I added some default classes to the pagination bullet div "swiper-pagination-bullets" and it seems like it's generating the bullets on the client side now. Previously with pure CSR code, this class was being added automatically.

**Comment by cernymatej** [comment]
Any updates on this? Is it possible to render the pagination on the server?
